sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/DoubleExemplar.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.data;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.Attributes;
+import javax.annotation.concurrent.Immutable;
+
+/** An {@link Exemplar} with {@code double} measurments. */
+@Immutable
+@AutoValue
+public abstract class DoubleExemplar implements Exemplar {
+
+  /**
+   * Construct a new exemplar.
+   *
+   * @param filteredAttributes The set of {@link Attributes} not already associated with the {@link
+   *     PointData}.
+   * @param recordTimeNanos The time when the sample qas recorded in nanoseconds.
+   * @param spanId (optional) The associated SpanId.
+   * @param traceId (optional) The associated TraceId.
+   * @param value The value recorded.
+   */
+  public static DoubleExemplar create(
```

Comment:
I'm not sure what do I walk into with this but would not a generic `Exemplar<T extends Number>` work?

Reply:
Maybe @jkwatson can comment on this?   I tried to match the current shape of the Java code, and Long vs. Double staying unboxed is a theme across all the metrics, so I stuck with it.  It has a natural pairing with the protocol buffers we plan to send this data as.

Reply:
We've shied away from using generic numbers for hot-path, high usage scenarios to avoid the extra object allocation required to store the information. Exemplars should be much lower volume, though, so this might be a case where we could gain readability/maintainability by eating an extra allocation to box the numbers.

Reply:
Alright, going to stick with existing patterns.